src/roadrunner-symfony-nyholm/src/Runner.php

```diff
@@ -30,13 +32,21 @@ class Runner implements RunnerInterface
      */
     private $sessionOptions;
 
-    public function __construct(KernelInterface $kernel, ?HttpFoundationFactoryInterface $httpFoundationFactory = null, ?HttpMessageFactoryInterface $httpMessageFactory = null)
+    public function __construct(HttpKernelInterface $kernel, ?HttpFoundationFactoryInterface $httpFoundationFactory = null, ?HttpMessageFactoryInterface $httpMessageFactory = null)
     {
         $this->kernel = $kernel;
         $this->psrFactory = new Psr7\Factory\Psr17Factory();
         $this->httpFoundationFactory = $httpFoundationFactory ?? new HttpFoundationFactory();
         $this->httpMessageFactory = $httpMessageFactory ?? new PsrHttpFactory($this->psrFactory, $this->psrFactory, $this->psrFactory, $this->psrFactory);
 
+        if ($kernel instanceof HttpCache) {
+            $kernel = $kernel->getKernel();
+        }
+
+        if (!$kernel instanceof KernelInterface) {
+            throw new \InvalidArgumentException(sprintf('Expected argument of type "%s", "%s" given.', KernelInterface::class, get_class($kernel)));
```

Comment:
`sprintf('Expected argument of type "%s" or "%s", "%s" given.', KernelInterface::class, HttpCache::class, get_class($kernel))`